bin/update-gh-pages-documentation-site

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2022 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+#
+# Updates the GitHub Pages documentation site thats published from the 'docs'
+# subdirectory in the 'gh-pages' branch of this repository.
+#
+# This script should be run by someone with commit access to the 'gh-pages' branch
+# at a regular frequency so that the documentation content on the GitHub Pages site
+# is up-to-date with the content in this repo.
+#
+
+set -eu
+
+# A `realpath` alternative using the default C implementation.
+filepath() {
+  [[ $1 = /* ]] && echo "$1" || echo "$PWD/${1#./}"
+}
+
+SWIFT_ASYNC_ALGORITHMS_ROOT="$(dirname $(dirname $(filepath $0)))"
+
+ASYNC_ALGORITHMS_BUILD_DIR="$SWIFT_ASYNC_ALGORITHMS_ROOT"/.build/async-algorithms-gh-pages-build
+ASYNC_SEQUENCE_VALIDATION_OUTPUT_DIR="$ASYNC_ALGORITHMS_BUILD_DIR"/async-sequence-validation-docs
+
+mkdir -p "$ASYNC_SEQUENCE_VALIDATION_OUTPUT_DIR"
+
+# Set current directory to the repository root
+cd "$SWIFT_ASYNC_ALGORITHMS_ROOT"
+
+# Use git worktree to checkout the gh-pages branch of this repository in a gh-pages sub-directory
+git fetch
+git worktree add --checkout gh-pages origin/gh-pages
+
+# Pretty print DocC JSON output so that it can be consistently diffed between commits
+export DOCC_JSON_PRETTYPRINT="YES"
+
+# Generate documentation for the 'AsyncAlgorithms' target and output it
+# to the /docs subdirectory in the gh-pages worktree directory.
+swift package \
+    --allow-writing-to-directory "$SWIFT_ASYNC_ALGORITHMS_ROOT/gh-pages/docs" \
+    generate-documentation \
+    --target AsyncAlgorithms \
+    --disable-indexing \
+    --transform-for-static-hosting \
+    --hosting-base-path swift-async-algorithms \
+    --output-path "$SWIFT_ASYNC_ALGORITHMS_ROOT/gh-pages/docs"
+
+# Generate documentation for the 'AsyncSequenceValidation' target and output it
+# to a temporary output directory in the .build directory.
+swift package \
+    --allow-writing-to-directory "$ASYNC_ALGORITHMS_BUILD_DIR" \
+    generate-documentation \
+    --target AsyncSequenceValidation \
```

Comment:
since this target is solely a testing artifact perhaps we should just skip it for now.

Reply:
Got it, I’ll update it.